network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/feature/unsafe_wifi/UnsafeWifiDetectionSettingView.kt

```diff
@@ -0,0 +1,151 @@
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.launch
+
+@InjectWith(ViewScope::class)
+class UnsafeWifiDetectionSettingView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+) : FrameLayout(context, attrs, defStyle) {
+
+    @Inject
+    lateinit var viewModelFactory: Factory
+
+    @Inject
+    lateinit var dispatcherProvider: DispatcherProvider
+
+    private var mainCoroutineScope: CoroutineScope? = null
+
+    // we use replay = 0 because we don't want to emit the last value upon subscription
+    private val events = MutableSharedFlow<Event>(replay = 1, extraBufferCapacity = 1)
+
+    private val binding: VpnViewSettingsUnsafeWifiBinding by viewBinding()
+
+    private val viewModel: UnsafeWifiDetectionViewModel by lazy {
+        ViewModelProvider(findViewTreeViewModelStoreOwner()!!, viewModelFactory)[UnsafeWifiDetectionViewModel::class.java]
+    }
+
+    private val toggleListener = OnCheckedChangeListener { p0, value ->
+        mainCoroutineScope?.launch {
+            events.emit(
+                if (value) OnEnableIntent else OnDisableIntent,
+            )
+        }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override fun onAttachedToWindow() {
+        AndroidSupportInjection.inject(this)
+        super.onAttachedToWindow()
+
+        @SuppressLint("NoHardcodedCoroutineDispatcher")
+        mainCoroutineScope = CoroutineScope(SupervisorJob() + dispatcherProvider.main())
+
+        binding.unsafeWifiDetection.setOnCheckedChangeListener(toggleListener)
+
+        ViewTreeLifecycleOwner.get(this)?.lifecycleScope?.launch {
+            events
+                .flatMapLatest { viewModel.reduce(it) }
+                .flowOn(dispatcherProvider.io())
+                .onStart { events.emit(Init) }
+                .collect(::render)
+        }
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        mainCoroutineScope?.cancel()
+        mainCoroutineScope = null
+    }
+
+    private fun render(state: State) {
+        when (state) {
+            Disable -> binding.unsafeWifiDetection.quietlySetIsChecked(false, toggleListener)
+            Enable -> binding.unsafeWifiDetection.quietlySetIsChecked(true, toggleListener)
+            Idle -> { /* no-op */
+            }
+        }
+    }
+
+    sealed class Event {
+        data object Init : Event()
+        data object OnEnableIntent : Event()
+        data object OnDisableIntent : Event()
+    }
+
+    sealed class State {
+        data object Idle : State()
+        data object Enable : State()
+        data object Disable : State()
+    }
+}
+
+@ContributesMultibinding(ActivityScope::class)
+@PriorityKey(UNSAFE_WIFI_DETECTION_PRIORITY)
+class UnsafeWifiDetectionSettingViewPlugin @Inject constructor(
+    private val appBuildConfig: AppBuildConfig,
+) : VpnSettingPlugin {
```

Comment:
@aitorvs i assume since we only want this in internal, we don’t want this VpnSettingPlugin here? or we add a debug check here?

Reply:
Not sureI understand, the `UnsafeWifiDetectionSettingViewPlugin` is inside the `network-protection-internal` module.

Reply:
Ahhhhh completely missed it sorry!